src/test/kotlin/com/metalbear/mirrord/MirrordPluginTest.kt

```diff
@@ -0,0 +1,213 @@
+            println("waiting for IDE...")
+            waitForIgnoringError(ofMinutes(3)) { remoteRobot.callJs("true") }
+        }
+
+        @AfterAll
+        @JvmStatic
+        fun cleanUp() {
+            val xpathPage = URL("http://localhost:8082/").readText()
+            // create a file in build/reports
+            val file = File("build/reports/robot-page.html")
+            file.writeText(xpathPage)
+            ideaProcess?.destroy()
+            tmpDir.toFile().deleteRecursively()
+        }
+    }
+
+    @Test
+    @Video
+    fun testMirrordFlow() = with(remoteRobot) {
+        step("Welcome Frame") {
+            welcomeFrame {
+                steps?.openProject(System.getProperty("test.workspace"))
+            }
+        }
+        idea {
+            step("Open `app.py`") {
+                // sometimes the projectViewTree does not open, this is quite random
+                // as a workaround we try to open it again
+                var clickToOpenProject = false
+                val projectViewTree = try {
+                    projectViewTree
+                } catch (e: Exception) {
+                    leftStripe {
+                        findText("Project").click()
+                    }
+                    clickToOpenProject = true
+                    projectViewTree
+                }
+                with(projectViewTree) {
+                    if (clickToOpenProject) {
+                        waitFor(ofSeconds(30)) {
+                            hasText("test-workspace")
+                        }
+                        findText("test-workspace").doubleClick()
+                    }
+                    waitFor(ofSeconds(30)) {
+                        hasText("app.py")
+                    }
+                    findText("app.py").doubleClick()
+                    editorTabs {
+                        waitFor {
+                            isFileOpened("app.py")
+                        }
+                    }
+                }
+                step("Set up Poetry Environment") {
+                    // blue hover appears on top of the text window asking
+                    // to set up poetry environment
+                    fileIntention {
+                        val setUpPoetry = setUpPoetry
+                        setUpPoetry.click()
+                        waitFor {
+                            !setUpPoetry.isShowing
+                        }
+                    }
+                    statusBar {
+                        // need to make sure poetry is not doing anything
+                        try {
+                            waitFor {
+                                !poetryProgress.isShowing
+                            }
+                        } catch (e: Exception) {
+                            // ignore
+                        }
+                    }
+                }
+
+                step("Enable mirrord and create config file") {
+                    waitFor(ofSeconds(30)) {
```

Comment:
Why are you batching them? Makes more sense to test the config first, see if it broke, and if it didn't move forward

Reply:
not sure what you mean? do you mean separate the button clicks?

Reply:
I mean separate steps - so you'd have a step for creating the config file, and we'd know that if that step failed, there's an issue with the config file creation.

Here's how I'd split it into steps:
1. create config file (this is mirrord functionality so should be tested)
2. open app.py, set breakpoint (this is just intellij functionality so nothing to test and can be batched)
3. enable mirrord, start debugging (enable mirrord shouldn't have its own step since it doesn't really have any functionality to test until you start debugging)